src/KlarnaSignIn.tsx

```diff
@@ -0,0 +1,41 @@
+import RNKlarnaSignIn from './specs/NativeKlarnaSignIn';
+
+export interface KlarnaSignInProps {
+  readonly environment: string;
+  readonly region: string;
+  readonly returnUrl: string;
+}
+
+// declare a class that receive the KlarnaSignInProps in the initializer and expose a method to sign in
+export class KlarnaSignIn {
+  readonly environment: string;
+  readonly region: string;
+  readonly returnUrl: string;
+
+  constructor(props: KlarnaSignInProps) {
+    this.environment = props.environment;
+    this.region = props.region;
+    this.returnUrl = props.returnUrl;
+    RNKlarnaSignIn.init(this.environment, this.region, this.returnUrl);
+  }
+
+  signIn(
+    clientId: string,
+    scope: string,
+    market: string,
+    locale: string,
+    tokenizationId: string
+  ): Promise<(result: any, error: any) => void> {
```

Comment:
A single fulfillment type for a promise would be better in this case rather a "callback" type of method